### PR TITLE
Updated font color to white

### LINKED ISSUE #5289: Datagrid: Tags text color is not that visible when in version new, dark mode
**Describe the bug**
Datagrid: Tags text color is not that visible when in version new, dark mode

**To Reproduce**

1. Go to https://main-enterprise.demo.design.infor.com/components/datagrid/example-alternate-row-shading.html?theme=new&mode=dark&colors=0066D4
2. Hover to each rows & notice the focused row tags text color is not that visible 

**Expected behavior**
Text should be in lighter color

**Version**
4.53 - ids - dev

**Screenshots**
image

**Platform**
All major browser & OS

**Additional context**
N/a

### PR DESCRIPTION
**Pull request by Jah25**
Updated the ids-color-palette to -white in the hover case.

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Previously the font on data grid tags would be black when the row was hovered over in dark mode making it extremely difficult to read. 

dark-tag

The font color has been updated to white to resolve this issue

light-tag

**Related github/jira issue (required)**:

Closes #5289 

**Steps necessary to review your pull request (required)**:

1) Pull this branch, build and run the application
2) navigate to here: http://localhost:4000/components/datagrid/example-alternate-row-shading.html?theme=new&mode=dark&colors=0066D4
3) hover over any of the data grid rows
4) tag background should change color, font color should remain white as depicted above.

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [ ] A note to the change log.

### CODE REVIEW

**Review by vivz753 — APPROVED** [review]

**Comment by ericangeles** [comment]
I see an issue here when you go to high contrast mode.

http://localhost:4000/components/datagrid/example-alternate-row-shading.html?theme=new&mode=contrast&colors=0066D4

high-contrast

If the issue only persists in dark mode, I suggest using the `theme-new-dark.scss` (for the new) and `theme-classic-dark.scss` for classic.

**Review by ericangeles — CHANGES_REQUESTED** [review]
I see an issue here when you go to high contrast mode.

http://localhost:4000/components/datagrid/example-alternate-row-shading.html?theme=new&mode=contrast&colors=0066D4

high-contrast

If the issue only persists in dark mode, I suggest using the `theme-new-dark.scss` (for the new) and `theme-classic-dark.scss` for classic.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - `[Monthview]` Fixed a bug where a vertical scroll is showing when it is unnecessary. ([#5350](https://github.com/infor-design/enterprise/issues/5350))
 - `[Multiselect]` Fixed a regression bug where clear icon were not properly aligned on compact mode. ([#5396](https://github.com/infor-design/enterprise/issues/5396))
 - `[Popdown]` Remove deprecation console warning. We still consider this component deprecated but will not remove until 5.0 version. The warning was only removed for now. ([#1070](https://github.com/infor-design/enterprise-ng/issues/1070))
+-`[Datagrid]` Fixed a bug where the font color on tags was black when a row was hovered over in dark mode. Font color now white. ([#5289] (https://github.com/infor-design/enterprise/issues/5289))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:21]:
```suggestion
-`[Datagrid]` Fixed a bug where the font color on tags was black when a row was hovered over in dark mode. Font color now white. ([#5289](https://github.com/infor-design/enterprise/issues/5289))
```

**Review by tmcconechy — CHANGES_REQUESTED** [review]
@Jah25 reviewing this behavior i think maybe we should fix it another way. 
Can you try to change it so that when you hover the that the tag color doesnt need to change (like the hyperlink column).

So i think to do that you just need to remove the `.tag` part of this nasty selector...

```
.datagrid > tbody > tr.is-hover-row:not(.error):not(.is-rowdisabled):not(.datagrid-virtual-row-bottom):not(.is-selected):not(.is-rowactivated):not(.datagrid-expandable-row) td:not(.is-editing):not(.is-readonly):not(.error) .tag:not(.info):not(.good):not(.error):not(.alert)
```

**Comment by Jah25** [comment]
@tmcconechy just to clarify, on hover neither the tag background color or text color will change.

**Comment by Jah25** [comment]
oh and should I keep the change for the azure version of the tag or revert that to using the ids-palette-white

**Comment by tmcconechy** [comment]
Correct, i'm thinking that makes more sense just the row background changes, tag background and text stays as it was.

**Comment by Jah25** [comment]
So in testing the new light theme I noticed some contrast issues with the background 

image

image

I'm also seeing similar issue with all the classic themes

image

image

image

Do we have leeway to experiment with the row highlight color or is that a locked in design decision?

**Comment by tmcconechy** [comment]
It would probably be ok to change the contrast highlight color a little but it can't be that drastic of a change. Looking at the screen shots i guess its not that bad but could be better on contrast themes.

This is really just a hover state for selection. But make a suggestion.

**Comment by tmcconechy** [comment]
Actually thinking about this probably better to fix this the minimal way possible (no design changes just fix the tag).

**Comment by Jah25** [comment]
got it so the desired change is: no change to tag background-color or font color on hover.

**Comment by ericangeles** [comment]
Suggestion: Maybe create a global variable name for the color tag when it hovers and the default color is white, then just change the value of it on dark mode, since the issue is only shown in that particular area.

**Comment by Jah25** [comment]
@ericangeles so for the font color I did add a variable $datagrid-badges-hover-font-color; that is defaulted to white in the config and then changed at the theme level.

**Comment by Jah25** [comment]
@tmcconechy since this has been merged into main should I leave it be or move forward with the requested change for the tag background color?

**Comment by tmcconechy** [comment]
@Jah25 that notification your seeing is just that someone merged main on your branch because something else was merged. Just go ahead and continue, nothing was merged into main yet

**Comment by tmcconechy** [comment]
Minor lint error on https://github.com/infor-design/enterprise/pull/5421/checks?check_run_id=3069038899#step:9:62

Also i see what you mean with the grey tags on Light/New Theme when you hover. This isnt a major issue because of the variations so what i would do for that is just update the example a little.

SO change this line https://github.com/infor-design/enterprise/blob/main/app/views/components/datagrid/example-alternate-row-shading.html#L29

```
ranges: [{'min': 151, 'max': 9999, 'classes': 'azure07'}, {'min': 0, 'max': 150, 'classes': 'slate03'}]
```

So basically change the example to show different colors. As this is only an issue in one single situation.

Also are both new variables used:

```
$datagrid-badges-hover-font-color: $ids-color-palette-black;
$datagrid-badges-azure-font-color: $ids-color-palette-white;
```

If im reading it correctly on one is?

**Comment by Jah25** [comment]
@tmcconechy oh yup thanks for pointing that out I forgot to pull the text color variable, I'll update that and update the example for the tag issue.

**Comment by Jah25** [comment]
Hey @tmcconechy I'm getting a fail on the build from popup menu button api that I didn't touch is this a known issue?

**Comment by tmcconechy** [comment]
A couple tests randomly fail. I've seen this one before. Did a rerun - this should fix

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Review by ericangeles — APPROVED** [review]
